src/odemis/driver/nfpm_netscan.py

```diff
@@ -48,16 +48,19 @@ def scan():
                 elif data == magic:
                     logging.debug("Skipping our own packet")
                     continue
-                elif data.startswith("\xfe") and len(data) > 25: # That should be an answer packet
+                elif data.startswith(b"\xfe") and len(data) > 25: # That should be an answer packet
                     try:
                         # Look for the hostname (default is like "8742-15433\x00")
-                        end_hn = data.index("\x00", 19)
+                        end_hn = data.index(b"\x00", 19)
                         hn = data[19:end_hn]
-                        print("%s\t%s\t%d" % (hn, addr, p))
+                        print(b"%s\t%s\t%d" % (hn, addr, p))
                     except Exception:
-                        logging.exception("Failed to decode packet %r from %s", data, addr)
+                        logging.exception("Failed to decode packet %r from %s",
+                                          data.decode("latin1", "backslashreplace"),
+                                          addr.decode("latin1", "backslashreplace"))
```

Comment:
I think that `addr`, as returned by `recvfrom()` is a string (also on Python 3). So there shouldn't be any `.decode()` needed... but you might want to double check (assuming that it's using IPv4, aka IF_INET.).